### PR TITLE
ENH: make tapkee compile even in case of multiple inclusions

### PR DESCRIPTION
**Pull request by YannLePoul**
There were missing many inline keywords in function implementation, some problems with static variable initialization (that usually needs to be in the cpp file), and a problem with way of using method traits to have compile time information on classes.

### CODE REVIEW

File: include/formatting/formatting.hpp [hunk 1]
```diff
@@ -408,37 +408,37 @@ namespace formatting
 		formatting::internal::ValueWrapperImplementationBase* implementation_;
 	};
 
-	std::string format(const std::string& fmt, 
+    inline std::string format(const std::string& fmt,
```
Comment (lisitsyn) [inline include/formatting/formatting.hpp]: It looks like it is <Tab>inline so these methods indentation is not consistent with other code.

File: include/tapkee/traits/methods_traits.hpp [hunk 2]
```diff
@@ -11,17 +11,16 @@ namespace tapkee
...
-	static const bool needs_distance;
-	static const bool needs_features;
-};
+template <int method, typename Enable = void> struct MethodTraits;
 
-#define METHOD_TRAIT(X,kernel_needed,distance_needed,features_needed)			\
-template <> const bool MethodTraits<X>::needs_kernel = kernel_needed;			\
-template <> const bool MethodTraits<X>::needs_distance = distance_needed;		\
-template <> const bool MethodTraits<X>::needs_features = features_needed		\
+#define METHOD_TRAIT(X,kernel_needed,distance_needed,features_needed)   \
```
Comment (lisitsyn) [inline include/tapkee/traits/methods_traits.hpp]: That's cool.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,22 @@
 project (Tapkee)
 
-cmake_minimum_required (VERSION 2.6.2)
+cmake_minimum_required (VERSION 2.8)
+
+include(CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
+CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
+if(COMPILER_SUPPORTS_CXX11)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```
Comment (lisitsyn) [inline CMakeLists.txt]: Could you please indent the body of if-else?

**Comment by lisitsyn** [comment]
Thanks a lot! Just a few minor issues then we can merge it.

File: include/tapkee/routines/matrix_operations.hpp [hunk 4]
```diff
@@ -184,11 +194,13 @@ struct GPUDenseImplicitSquareMatrixOperation
 	viennacl::matrix<ScalarType> mat;
 	viennacl::vector<ScalarType> vec;
 	viennacl::vector<ScalarType> res;
-	static const char* ARPACK_CODE;
-	static const bool largest;
+    static const std::string& ARPACK_CODE()
```
Comment (lisitsyn) [inline include/tapkee/routines/matrix_operations.hpp]: One more thing! Sorry :)

File: include/tapkee/routines/matrix_operations.hpp [hunk 5]
```diff
@@ -215,11 +227,13 @@ struct GPUDenseMatrixOperation
...
```
Comment (lisitsyn) [inline include/tapkee/routines/matrix_operations.hpp]: One more thing

Comment (YannLePoul) [inline include/tapkee/routines/matrix_operations.hpp]: the fight of tabs vs spaces !

**Comment by lisitsyn** [comment]
Thanks again!

**Comment by YannLePoul** [comment]
Happy to help !

**Pull request merged**